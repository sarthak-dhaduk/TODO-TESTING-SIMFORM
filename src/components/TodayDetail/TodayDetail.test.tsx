import { render, screen } from "@testing-library/react";
import TodayDetail, { days, monthArr } from "./TodayDetail";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

describe("TodayDetail component", () => {
  //testing the ui snapshot
  test("TodaysDetail renders correctly", () => {
    const renderedComponent = renderer.create(<TodayDetail />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  test("renders today's date, month, year and day", () => {
    const currentDate = new Date();
    const date = currentDate.getDate();
    const month = monthArr[currentDate.getMonth()];
    const year = currentDate.getFullYear();
    const day = days[currentDate.getDay()];

    render(<TodayDetail />);
    const dateElement = screen.getByText(date);
    const monthElement = screen.getByText(month);
    const yearElement = screen.getByText(year);
    const dayElement = screen.getByText(day);

    expect(dateElement).toBeInTheDocument();
    expect(monthElement).toBeInTheDocument();
    expect(yearElement).toBeInTheDocument();
    expect(dayElement).toBeInTheDocument();
  });
});
