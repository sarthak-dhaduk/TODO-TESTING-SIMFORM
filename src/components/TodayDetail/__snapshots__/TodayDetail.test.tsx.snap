// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodayDetail component TodaysDetail renders correctly 1`] = `
<div
  className="flex justify-between my-4 mt-12 items-center"
>
  <div
    className="flex"
  >
    <div
      className="text-5xl mr-3 font-medium"
    >
      19
    </div>
    <div
      className="flex flex-col"
    >
      <div
        className="font-medium"
      >
        APRIL
      </div>
      2023
    </div>
  </div>
  <div
    className="font-medium"
  >
    WEDNESDAY
  </div>
</div>
`;
